src/v2/view-builder/views/IdentifierView.js

```diff
@@ -19,17 +22,38 @@ const Body = BaseForm.extend({
   render() {
     BaseForm.prototype.render.apply(this, arguments);
 
-    // Launch Device Authenticator
-    if (this.options.appState.hasRemediationObject(RemediationForms.LAUNCH_AUTHENTICATOR)) {
-      this.add(signInWithDeviceOption, '.o-form-fieldset-container', false, true, { isRequired: false });
-    }
-
+    const { appState, settings } = this.options;
+    const hasLaunchAuthenticator = appState.hasRemediationObject(RemediationForms.LAUNCH_AUTHENTICATOR);
+    
     // This IdentifierView has been reused for the case when there is no `identify` remediation form
     // but only `redirect-idp` forms. At that case, no UI Schema.
     const hasUISchemas = this.getUISchema().length > 0;
+    const idpButtons = createIdpButtons(appState.get('remediations'));
+    const customButtons = createCustomButtons(settings);
+
+    if (!hasUISchemas) {
+      // redirect-idp
+      this.$el.find('.button-primary').hide();
+
+      // Special case: only one IDP. may auto-redirect
+      const onlyOneIdp = idpButtons.length === 1 && customButtons.length === 0 && !hasLaunchAuthenticator;
+      const isOAuth = settings.get('oauth2Enabled');
+      const isInitialResponse = appState.get('responseCount') <= 0;
+      const shouldRedirect = onlyOneIdp && (isOAuth ? !isInitialResponse : true);
```

Comment:
I think it could be 
`onlyOneIdp && (!isOAuth || !isInitialResponse)`

Bettor to have comments to explain. I'm using True/False table to help me understand it.

```
| onlyOneIdp | isOAuth | isInitialResponse | redirect |
|------------+---------+-------------------+----------|
| Y          | Y       | Y                 | N        |
| Y          | Y       | N                 | Y        |
| Y          | N       | Y                 | Y        |
| Y          | N       | N                 | Y        |
| N          | Y       | Y                 | N        |
| N          | Y       | N                 | N        |
| N          | N       | Y                 | N        |
| N          | N       | N                 | N        |
```